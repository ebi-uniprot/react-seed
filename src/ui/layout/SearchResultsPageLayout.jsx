
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import DefaultPageContent from './DefaultPageContent';

const SearchResultsPageLayout = props => (
  <div className="search-results-page-layout">
    <b>Search Results: </b>
    <Link to="/">Go back to Home page</Link>

    <DefaultPageContent>
      {props.content}
    </DefaultPageContent>
  </div>
);

SearchResultsPageLayout.propTypes = {
  content: PropTypes.element,
};

SearchResultsPageLayout.defaultProps = {
  content: () => <h3>Page Content</h3>,
};

export default SearchResultsPageLayout;
