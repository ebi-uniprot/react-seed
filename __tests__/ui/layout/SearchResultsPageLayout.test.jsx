
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import SearchResultsPageLayout from '../../../src/ui/layout/SearchResultsPageLayout';

describe('SearchResultsPageLayout component', () => {
  test('should render', () => {
    let component = (
      <MemoryRouter>
        <SearchResultsPageLayout
          content={<h3>Works</h3>}
        />
      </MemoryRouter>
    );

    component = renderer
      .create(component)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
