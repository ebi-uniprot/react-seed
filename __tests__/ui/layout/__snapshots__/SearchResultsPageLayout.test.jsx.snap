// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsPageLayout component should render 1`] = `
<div
  className="search-results-page-layout"
>
  <b>
    Search Results: 
  </b>
  <a
    href="/"
    onClick={[Function]}
  >
    Go back to Home page
  </a>
  <div
    className="default-page-conent"
  >
    <h3>
      Works
    </h3>
  </div>
</div>
`;
